functions/src/index.ts

```diff
@@ -144,3 +153,86 @@ export const executeIndexOperation = functions.handler.firestore.document
       }
     }
   });
+
+export const executeFullIndexOperation = functions.tasks
+  .taskQueue()
+  .onDispatch(async (data: any) => {
+    const runtime = getExtensions().runtime();
+    if (!config.doFullIndexing) {
+      await runtime.setProcessingState(
+        'PROCESSING_COMPLETE',
+        'Existing documents were not indexed because "Indexing existing documents?" is configured to false. ' +
+        'If you want to run a full reindex, reconfigure this instance.'
+      );
+      return;
+    }
+
+    const offset = (data['offset'] as number) ?? 0;
+    const pastSuccessCount = (data['successCount'] as number) ?? 0;
+    const pastErrorCount = (data['errorCount'] as number) ?? 0;
+    // We also track the start time of the first invocation, so that we can report the full length at the end.
+    const startTime = (data['startTime'] as number) ?? Date.now();
+
+    const snapshot = await admin
+      .firestore()
+      .collection(process.env.COLLECTION_PATH)
+      .offset(offset)
```

Comment:
Using offsets incurs additional costs:

> However, when you send a query that includes an offset, you are charged a read for each skipped document. For example, if your query uses an offset of 10, and the query returns 1 document, you are charged for 11 reads. Because of this additional cost, you should use cursors instead of offsets whenever possible. &mdash; https://firebase.google.com/docs/firestore/pricing

Please refactor to use cursors, if possible.